addons/project/models/project.py

```diff
@@ -72,7 +72,7 @@ def _get_default_project_ids(self):
     description = fields.Text(translate=True)
     sequence = fields.Integer(default=1)
     project_ids = fields.Many2many('project.project', 'project_task_type_rel', 'type_id', 'project_id', string='Projects',
-        default=_get_default_project_ids)
+        default=_get_default_project_ids, required=True)
```

Comment:
@ppr-odoo
I think you are right (it adds an SQL constraint and therefore it is a schema change that is not possible).

I've modified it so that it is only required when you are on the form view of the `project.task.type` model.